.github/workflows/build.yml

```diff
@@ -0,0 +1,34 @@
+name: Update the build branch
+
+on:
+  push:
+    branches: [ 'trunk' ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Build and push to branch
+    steps:
+      - name: git-checkout
+        uses: actions/checkout@v2
+
+      - name: Install all dependencies
+        run: yarn run initial-setup
+
+      - name: Build
+        run: yarn workspaces run build
+
+      # Got to be a better way than this, but it works.
+      - name: Remove Source files
+        run: |
+          rm -rf source/wp-content/plugins/wporg-internal-notes/src
+          rm -rf source/wp-content/plugins/wporg-internal-notes/node_modules
+
+      - name: Push
+        uses: s0/git-publish-subdir-action@develop
+        env:
+          REPO: self
+          BRANCH: build
+          FOLDER: source/wp-content/plugins/wporg-internal-notes
```

Comment:
Seems like the "build" version should only have the plugin files in it, and none of the composer/package/env stuff. I.e. the contents of `source/wp-content/plugins/wporg-internal-notes` should be at the root. Is that what this line does?

Reply:
Yep :)

Here's my WIP of making this action: https://github.com/WordPress/wporg-internal-notes/compare/trunk...dd32:trunk & here's an example of the result:  https://github.com/dd32/wporg-internal-notes/tree/build

I reverted deleting package.json as I thought keeping it might allow composer to use it better, but.. it still failed.. so I'm going to have to defer to someone else on the proper package.json for that branch.

Reply:
That's also why the previous line in the action deletes folders from the plugin folder. The alternative was to copy the wanted files to a different directory and then push it to the branch, but deleting was easier since the build is done..

Reply:
I don't think you need to keep package.json, as that's all just stuff for building the source, which we're removing in the build branch anyway. In order to use it with composer as a VCS repository, though, you might need to add in a simple composer.json file. It wouldn't be the same as the one in the root of this repo, since again you don't need all the build tools and testing dependencies (like other WP plugins). It probably just needs something like this:

```json
{
	"name": "wporg/wporg-internal-notes",
	"description": "",
	"homepage": "https://wordpress.org",
	"license": "GPL-2.0-or-later",
	"support": {
		"issues": "https://github.com/WordPress/wporg-internal-notes/issues"
	}
}
```

Reply:
Thanks @coreymckrill!

I've added it in the PR, you can see the built example here: https://github.com/dd32/wporg-internal-notes/blob/build/package.json

It's just a sub-set of the fields included in the current package.json